pkg/scheduler/framework/preemption/preemption_test.go

```diff
@@ -354,3 +366,90 @@ func TestSelectCandidate(t *testing.T) {
 		})
 	}
 }
+
+func TestFindCandidates(t *testing.T) {
+	tests := []struct {
+		name               string
+		allNodes           []*framework.NodeInfo
+		pod                *v1.Pod
+		node               *v1.Node
+		nodeStatuses       map[string]*framework.Status
+		expectedError      bool
+		expectedCandidates int
+	}{
+		{
+			name:               "no available nodes",
+			allNodes:           []*framework.NodeInfo{},
+			pod:                st.MakePod().Name("test-pod").Priority(highPriority).Obj(),
+			node:               st.MakeNode().Name("").Obj(),
+			nodeStatuses:       map[string]*framework.Status{},
+			expectedError:      true,
+			expectedCandidates: 0,
+		},
+		{
+			name: "no potential candidates",
+			allNodes: []*framework.NodeInfo{
+				framework.NewNodeInfo(st.MakePod().Name("victim-pod").Node("node1").Priority(midPriority).Obj()),
+			},
+			pod:  st.MakePod().Name("test-pod").Priority(highPriority).Obj(),
+			node: st.MakeNode().Name("node1").Obj(),
+			nodeStatuses: map[string]*framework.Status{
+				"node1": framework.NewStatus(framework.Success),
+			},
+			expectedError:      false,
+			expectedCandidates: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			nodeToStatus := &FakeNodeToStatusReader{nodes: tt.nodeStatuses}
+
+			nodes := []*v1.Node{tt.node}
+
+			client := clientsetfake.NewClientset()
+			snap := internalcache.NewSnapshot([]*v1.Pod{tt.pod}, nodes)
+			_, ctx := ktesting.NewTestContext(t)
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			var objs = []runtime.Object{tt.pod}
+
+			for _, n := range nodes {
+				objs = append(objs, n)
+			}
+
+			informerFactory := informers.NewSharedInformerFactory(clientsetfake.NewClientset(objs...), 0)
+			registeredPlugins := append([]tf.RegisterPluginFunc{
+				tf.RegisterQueueSortPlugin(queuesort.Name, queuesort.New)},
+				tf.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+			)
+
+			fwk, err := tf.NewFramework(
+				ctx,
+				registeredPlugins, "",
+				frameworkruntime.WithPodNominator(internalqueue.NewSchedulingQueue(nil, informerFactory)),
+				frameworkruntime.WithInformerFactory(informerFactory),
+				frameworkruntime.WithClientSet(client),
+				frameworkruntime.WithSnapshotSharedLister(snap),
+			)
+
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			ev := Evaluator{
+				Handler: fwk,
+			}
+
+			gotCandidates, _, err := ev.findCandidates(ctx, tt.allNodes, tt.pod, nodeToStatus)
```

Comment:
Talked offline, we can mock the offset with `FakePreemptionScorePostFilterPlugin`.

However, to be host, although we **named** the function as `TestFindCandidates` but seems we only tested the `NodesForStatusCode`, which is already covered in `pkg/scheduler/framework/interface_test.go`, and the core logic of findCandidates is `DryRunPreemption` which also be tested in other places. So I think it's not a good idea to test this private function, which will add additional maintenance cost but values very little, instead, I would like to see more testcases in **TestPreempt**.